src/test/fuzz/util/descriptor.cpp

```diff
@@ -84,3 +87,34 @@ bool HasDeepDerivPath(const FuzzBufferType& buff, const int max_depth)
     }
     return false;
 }
+
+bool HasLargeFrag(const FuzzBufferType& buff, const int max_subs)
+{
+    std::stack<int> counts;
+    for (const auto& ch: buff) {
+        if (ch == '(') {
+            counts.push(1);
+        } else if (ch == ',' && !counts.empty()) {
+            if (++counts.top() > max_subs) return true;
+        } else if (ch == ')' && !counts.empty()) {
+            counts.pop();
+        }
+    }
+    return false;
+}
+
+bool HasTooManyWrappers(const FuzzBufferType& buff, const int max_wrappers)
+{
+    auto count{0};
+    // We iterate in reverse order to start counting when we encounter a colon.
+    for (const auto ch: reverse_iterate(buff)) {
+        if (ch == ':') {
+            count++;
```

Comment:
developer-notes-nit:
```suggestion
            ++count;
```

Reply:
Arg! It slipped through. Will address as i'll retouch.